parachain/pallets/control/src/lib.rs

```diff
@@ -131,31 +156,156 @@ pub mod pallet {
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(origin: OriginFor<T>) -> DispatchResult {
-			let mut location: MultiLocation = T::CreateAgentOrigin::ensure_origin(origin)?;
+			let origin_location: MultiLocation = T::ControlOrigin::ensure_origin(origin)?;
 
-			// Normalize all locations relative to the relay chain unless its the relay itself.
-			let relay_location = T::RelayLocation::get();
-			if location != relay_location {
-				location
-					.reanchor(&relay_location, T::UniversalLocation::get())
-					.or(Err(Error::<T>::LocationConversionFailed))?;
-			}
+			let (agent_id, para_id, location) = Self::convert_location(origin_location)?;
 
-			// Hash the location to produce an agent id
-			let agent_id = T::AgentHashedDescription::convert_location(&location)
-				.ok_or(Error::<T>::LocationConversionFailed)?;
+			log::debug!(
+				target: LOG_TARGET,
+				"💫 Create Agent request with agent_id {:?}, origin_location at {:?}, location at {:?}",
+				agent_id,
+				origin_location,
+				location
+			);
 
 			// Record the agent id or fail if it has already been created
-			if Agents::<T>::get(agent_id).is_some() {
-				return Err(Error::<T>::AgentAlreadyCreated.into())
-			}
-			Agents::<T>::insert(agent_id, ());
+			ensure!(!Agents::<T>::contains_key(agent_id), Error::<T>::AgentAlreadyCreated);
+
+			Agents::<T>::insert(
+				agent_id,
+				AgentInfo { origin: VersionedMultiLocation::from(location), para_id },
+			);
+
+			let message = Message {
+				origin: T::OwnParaId::get(),
+				command: Command::CreateAgent { agent_id, para_id },
```

Comment:
It's not really useful to include the parachain id when creating the agent.

See my comment above.

Reply:
https://github.com/Snowfork/snowbridge/pull/917#discussion_r1287119996